pkg/cluster/audit.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"os"
+
+	"github.com/coreos/etcd-operator/pkg/spec"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+func NewAuditLogger(cl *spec.Cluster, lg *logrus.Entry) *logrus.Logger {
+	if cl.Spec.SelfHosted == nil {
+		return nil
+	}
+
+	mountPath := "/var/tmp/etcd-operator/"
+	_, err := os.Stat(mountPath)
+	if os.IsNotExist(err) {
+		lg.Infof("mountPath(%v) not detected, no auditing will be performed: %v", mountPath, err)
+		return nil
+	}
+	lg.Infof("detected the mountPath(%v): starting to audit operator actions to the mountPath", mountPath)
+
+	fileName := mountPath + cl.Metadata.Name + ".log"
+	logFile, err := os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+	if err != nil {
+		lg.Errorf("failed to open logfile(%v): %v", fileName, err)
+		return nil
+	}
+
+	l := logrus.New()
+	l.Out = logFile
+	l.Infof("Starting audit logs for self-hosted etcd cluster: %v", cl.Metadata.Name)
+	return l
+}
+
+func (c *Cluster) auditPodCreation(pod *v1.Pod, podCreationErr error) {
```

Comment:
We should provide an AuditLogger object instead of coupling it as Cluster methods.